x-pack/plugins/ml/server/routes/data_frame_analytics.ts

```diff
@@ -13,8 +14,55 @@ import {
   dataAnalyticsExplainSchema,
   analyticsIdSchema,
   stopsDataFrameAnalyticsJobQuerySchema,
+  analyticsIdIndexSchema,
 } from './schemas/data_analytics_schema';
+import { IndexPatternHandler } from '../models/data_frame_analytics/index_patterns';
 
+function getIndexPatternId(context: RequestHandlerContext, patternName: string) {
+  const iph = new IndexPatternHandler(
+    context.ml!.mlClient.callAsCurrentUser,
+    context.core.savedObjects.client
+  );
+  return iph.getIndexPatternId(patternName);
+}
+
+function deleteIndexPatternById(context: RequestHandlerContext, indexPatternId: string) {
+  const iph = new IndexPatternHandler(
+    context.ml!.mlClient.callAsCurrentUser,
+    context.core.savedObjects.client
+  );
+  return iph.deleteIndexPatternById(indexPatternId);
+}
+
+async function userCanDeleteIndex(
+  context: RequestHandlerContext,
+  destinationIndex: string
+): Promise<boolean> {
+  let privilege;
+  try {
+    privilege = await context.ml!.mlClient.callAsCurrentUser('transport.request', {
```

Comment:
this can be `callAsCurrentUser('ml.privilegeCheck', { body: ... });`

Reply:
Switched call to `ml.privilgeCheck` in  https://github.com/elastic/kibana/pull/66934/commits/d14fafab86d64ef3afce4bda0f3c415ea85e672e